fineract-provider/src/main/java/org/apache/fineract/portfolio/loanaccount/domain/Loan.java

```diff
@@ -3893,6 +3893,23 @@ public ChangedTransactionDetail close(final JsonCommand command, final LoanLifec
         return changedTransactionDetail;
     }
 
+    public void handleChargebackTransaction(LoanTransaction chargebackTransaction,
+            final LoanLifecycleStateMachine loanLifecycleStateMachine) {
+        chargebackTransaction.updateLoan(this);
+
+        final LoanRepaymentScheduleTransactionProcessor loanRepaymentScheduleTransactionProcessor = this.transactionProcessorFactory
+                .determineProcessor(this.transactionProcessingStrategy);
+
+        loanRepaymentScheduleTransactionProcessor.handleRefund(chargebackTransaction, getCurrency(), getRepaymentScheduleInstallments(),
```

Comment:
overpaid portion is unhandled
E.g.:
Disbursed amount is 1000, outstanding is 1000
Repaid 700, outstanding is 300
Chargeback 700, outstanding should be 0 and overpaid by 400